tests/testthat/helper-db.R

```diff
@@ -0,0 +1,88 @@
+get_db_info <- function() {
+  list(
+    dbname = "vimpact_test_db",
+    user = "postgres",
+    host = "localhost"
+  )
+}
+
+prepare_example_postgres_db <- function() {
+  info <- get_db_info()
+  vimpact_test_postgres_connection(info$dbname, info$user, info$host)
+  con <- get_test_connection()
+  add_dummy_data(con)
+  invisible(TRUE)
+}
+
+get_test_connection <- function() {
+  info <- get_db_info()
+  con <- get_postgres_connection(info$dbname, info$user, info$host)
+  withr::defer_parent(DBI::dbDisconnect(con))
+  con
+}
+
+vimpact_test_postgres_connection <- function(dbname, user, host) {
+  tryCatch(
+    get_postgres_connection(dbname, user, host),
+    error = function(e) testthat::skip(sprintf(
+    "Failed to open db connection to postgres db %s with user %s and host %s.",
+    dbname, user, host))
+  )
+}
+
+get_postgres_connection <- function(dbname, user, host) {
+  DBI::dbConnect(RPostgres::Postgres(), dbname = dbname, user = user,
+                 host = host)
+}
+
+add_dummy_data <- function(con) {
+  message("Adding dummy data to database")
```

Comment:
Can you please add a check here to make sure that this is never run on annex itself? I think that you can check `con` to find out if it's localhost (even if you just `format` it and look for `127.0.0.1` or equivalent, but possibly by looking at its slots - try `slotNames(con)`).

It's unlikely but this function might accidentally do a nasty delete on annex. I presume that ordinarily we have only a readonly user though? Still `DROP TABLE...CASCADE` is a scary looking command!

Reply:
Added a check on the db name - if it doesn't match `vimpact_test_db` then we throw an error